### ISSUE TITLE
An empty sequence is not allowed as the value of parameter $href [DOT 2.3.1]

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2432]
Using one of my client's samples I obtain an error when publishing to XHTML-based outputs, the error is something like this:

```
        maplink:
          [maplink] Error on line 418 of maplinkImpl.xsl:
          [maplink]   XPTY0004: An empty sequence is not allowed as the value of parameter $href
          [maplink]   at xsl:apply-templates (file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/xsl/preprocess/maplinkImpl.xsl#410)
          [maplink]      processing /map/reltable[1]/relheader[1]/relcolspec[1]/topichead[1]
          [maplink]   at xsl:apply-templates (file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/xsl/preprocess/maplinkImpl.xsl#335)
```

and it occurs at line:

```
    <xsl:template match="*[contains(@class, ' map/topicref ')]" mode="grab-group-title" as="xs:string?">
      <xsl:variable name="file-origin">
        <xsl:call-template name="get-file-uri">
          <xsl:with-param name="href" select="@href"/>
```

so somehow the topicref does not have an @href attribute set to it and the publishing breaks.
I added an xsl:message and it breaks when processing this topichead:

```
        <topichead xmlns:ditaarch="http://dita.oasis-open.org/architecture/2005/" xmlns:dita-ot="http://dita-ot.sourceforge.net/ns/201007/dita-ot" class="+ map/topicref mapgroup-d/topichead " xtrc="topichead:1;28:28" xtrf="file:/C:/Users/radu_coravu/Desktop/test/abbreviated/Producer_UG.ditamap" toc="no" print="no"><topicmeta class="- map/topicmeta " xtrc="topicmeta:2;29:32" xtrf="file:/C:/Users/radu_coravu/Desktop/test/abbreviated/Producer_UG.ditamap"><navtitle class="- topic/navtitle " xtrc="navtitle:1;30:35" xtrf="file:/C:/Users/radu_coravu/Desktop/test/abbreviated/Producer_UG.ditamap">See also</navtitle></topicmeta></topichead>
```

**Comment by raducoravu** [comment]
I'm not sure I can provide a sample for this, I'm testing on a very large confidential project, maybe this can be fixed without providing a sample though.

**Comment by eerohele** [comment]
[Here's](https://github.com/eerohele/dita-ot-issues/tree/master/fixtures/2432) a minimal set of files for reproducing the issue.

**Comment by raducoravu** [comment]
@eerohele saves the day :+1: 
By the way, this does not occur with DITA OT 1.8.

**Comment by eerohele** [comment]
I think the simplest fix is:

```diff
diff --git a/src/main/xsl/preprocess/maplinkImpl.xsl b/src/main/xsl/preprocess/maplinkImpl.xsl
index 91c67f2..120a51a 100644
--- a/src/main/xsl/preprocess/maplinkImpl.xsl
+++ b/src/main/xsl/preprocess/maplinkImpl.xsl
@@ -412,7 +412,7 @@
     </xsl:choose>
   </xsl:template>

-  <xsl:template match="*[contains(@class, ' map/topicref ')]" mode="grab-group-title" as="xs:string?">
+  <xsl:template match="*[contains(@class, ' map/topicref ')][normalize-space(@href)]" mode="grab-group-title" as="xs:string?">
     <xsl:variable name="file-origin">
       <xsl:call-template name="get-file-uri">
         <xsl:with-param name="href" select="@href"/>
```

@raducoravu: Can you try that and see whether it gives the correct result?

**Comment by raducoravu** [comment]
I think that the template should still be entered even when we have a topic head because the template has some <xsl:when> clauses, one of which which will gather the navtitle from the topichead.
So maybe the template should be rewritten like this:

        <xsl:template match="*[contains(@class, ' map/topicref ')]" mode="grab-group-title" as="xs:string?">
          <xsl:choose>
            <xsl:when test="parent::*[contains(@class, ' map/relcolspec ')]/*[contains(@class, ' topic/title ')][not(title = '')]">
              <xsl:value-of select="parent::*[contains(@class, ' map/relcolspec ')]/*[contains(@class, ' topic/title ')]"/>
            </xsl:when>
            <xsl:when test="descendant::*[contains(@class,' map/topicmeta ')]/*[contains(@class, ' topic/navtitle ')]">
              <xsl:value-of select="descendant::*[contains(@class,' map/topicmeta ')]/*[contains(@class, ' topic/navtitle ')]"/>
            </xsl:when>
            <xsl:when test="@navtitle and not(@navtitle = '')">
              <xsl:value-of select="@navtitle"/>
            </xsl:when>
            <xsl:otherwise>
              <xsl:if test="@href">
                <xsl:variable name="file-origin">
                  <xsl:call-template name="get-file-uri">
                    <xsl:with-param name="href" select="@href"/>
                    <xsl:with-param name="file-prefix" select="$file-prefix"/>
                  </xsl:call-template>
                </xsl:variable>
                <xsl:variable name="file">
                  <xsl:call-template name="replace-blank">
                    <xsl:with-param name="file-origin">
                      <xsl:value-of select="$file-origin"/>
                    </xsl:with-param>
                  </xsl:call-template>
                </xsl:variable>
                <xsl:if test="document($file,/)//*[contains(@class, ' topic/title ')]">
                  <xsl:value-of select="document($file,/)//*[contains(@class, ' topic/title ')][1]"/>
                </xsl:if>
              </xsl:if>
            </xsl:otherwise>
          </xsl:choose>
        </xsl:template>

**Comment by eerohele** [comment]
Ah, yes, your proposal is correct — upon closer inspection, that my suggestion happened to produce the correct result in this case was a coincidence.

**Comment by robander** [comment]
Fixed with #2873
